pkg/operators/wasm/wasm.go

```diff
@@ -94,6 +104,68 @@ func (i *wasmOperatorInstance) ExtraParams(gadgetCtx operators.GadgetContext) ap
 	return nil
 }
 
+func (i *wasmOperatorInstance) addHandle(obj any) uint32 {
+	if obj == nil {
+		return 0
+	}
+
+	i.handleLock.Lock()
+	defer i.handleLock.Unlock()
+
+	if len(i.handleMap) == math.MaxUint32-1 {
+		i.logger.Warnf("handle map is full")
+		return 0
+	}
+
+	handleIndex := i.lastHandleIndex
+	handleIndex++
+
+	// look for a free index in the map
+	for {
+		// zero is reserved
+		if handleIndex == 0 {
+			handleIndex++
+		}
+
+		if _, ok := i.handleMap[handleIndex]; !ok {
+			// register new entry
+			i.handleMap[handleIndex] = obj
+			i.lastHandleIndex = handleIndex
+			return handleIndex
+		}
+		handleIndex++
+	}
```

Comment:
> Despite having a map, you are looping on index, like an array, so you are doing O(n)

I think it's only O(n) in the worst case when the map is almost full and it has to check all possible combinations before finding the only available slot. In the first 2^32 calls to addHandle() the algorithm will return in O(1). 

> Instead of using this index, cannot you generate some random number?

I thought about this idea but I found some issues:
- The returned handled will be random, I think it's easier to understand sequential handles (like file descriptors in linux).
- If there's a collision we'll have to retry. If the map has few available slots, it'll require a lot of retries. Even if I think it shouldn't be a real issue given that wasm modules should use only few handles.

Reply:
TBH I think the current implementation works fine and we don't need to complicate it anymore. I looked at the implementation on Linux and AFAIU it's more complex, they for sure need to handle scalability problems we don't have. 

Something I can make, is to limit the maximum number of opened handles a gadget can have, something like 4k should be more than enough.